File: Sources/Shared/WaveRoom.ts
/** Represents the current state of a wave room */
export enum WaveRoomState {
    // State waiting, mean waiting for players
    WAITING,
    // State started, mean playing game currently
    PLAYING,
    // State ended, mean already game over
    ENDED,
}

/** Determines if a wave room is public or private */
export enum WaveRoomVisibleState {
    PUBLIC,
    PRIVATE,
}

export const VISIBLE_STATE_VALUES = Object.values(WaveRoomVisibleState);

/** Indicates if a player is ready to start the wave */
export enum WaveRoomPlayerReadyState {
    PREPARING,
    READY,
}

export const PLAYER_STATE_VALUES = Object.values(WaveRoomPlayerReadyState);

File: Sources/Client/private/src/Game/UI/Shared/Wave/WaveRoom.zig
pub const PlayerId = u16;

/// State of a wave room.
pub const State = enum(u8) {
    /// Waiting, mean waiting for players (not starting).
    waiting,
    /// Playing, mean playing currently.
    playing,
    /// Ended, mean game is already over.
    ended,
};

/// Visibility of a wave room.
pub const Visibility = enum(u8) {
    public,
    private,
};

/// Player ready state in a wave room.
pub const PlayerReadyState = enum(u8) {
    preparing,
    ready,
};
